File: lib/components/group_card.dart
import 'package:flutter/material.dart';
import '../src/global_styles.dart';
import '../src/bouncy_button.dart';

enum GroupStatus { active, paymentProcess, done }

class GroupModel {
  final String id;
  final String name;
  final String? avatarUrl;
  final GroupStatus status;
  final List<String> memberAvatars; 
  final int totalMemberCount; 
  final Color? avatarColor;

  GroupModel({
    required this.id,
    required this.name,
    this.avatarUrl,
    required this.status,
    List<String>? memberAvatars, 
    this.totalMemberCount = 0, 
    this.avatarColor,
  }) : memberAvatars = memberAvatars ?? [];

  int get memberCount => totalMemberCount;
  
  List<String> get displayAvatars => memberAvatars.take(4).toList();
}

class GroupCard extends StatelessWidget {
  final GroupModel group;
  final VoidCallback? onTap;

  const GroupCard({
    Key? key,
    required this.group,
    this.onTap,
  }) : super(key: key);

  String _getStatusText(GroupStatus status) {
    switch (status) {
      case GroupStatus.active:
        return 'Active';
      case GroupStatus.paymentProcess:
        return 'Bill Sharing';
      case GroupStatus.done:
        return 'Done';
    }
  }

  Color _getStatusTextColor(GroupStatus status) {
    switch (status) {
      case GroupStatus.active:
        return AppColors.active; 
      case GroupStatus.paymentProcess:
        return AppColors.onProcess; 
      case GroupStatus.done:
        return AppColors.done;
    }
  }

  Color _getStatusBackgroundColor(GroupStatus status) {
    switch (status) {
      case GroupStatus.active:
        return AppColors.active.withOpacity(0.1); 
      case GroupStatus.paymentProcess:
        return AppColors.onProcess.withOpacity(0.1); 
      case GroupStatus.done:
        return AppColors.done.withOpacity(0.1);
    }
  }

  Color _getStatusBorderColor(GroupStatus status) {
    switch (status) {
      case GroupStatus.active:
        return AppColors.active.withOpacity(0.3); 
      case GroupStatus.paymentProcess:
        return AppColors.onProcess.withOpacity(0.3); 
      case GroupStatus.done:
        return AppColors.done.withOpacity(0.3);
    }
  }

  Widget _buildMemberAvatars() {
    if (group.memberAvatars.isEmpty) {
      return SizedBox.shrink();
    }

    if (group.memberAvatars.length == 1) {
      return Container(
        width: 32,
        height: 32,
        decoration: BoxDecoration(
          shape: BoxShape.circle,
          border: Border.all(color: Colors.white, width: 2),
        ),
        child: ClipOval(
          child: group.memberAvatars[0].isNotEmpty
              ? Image.asset(
                  group.memberAvatars[0],
                  fit: BoxFit.cover,
                  errorBuilder: (context, error, stackTrace) {
                    return Container(
                      color: Colors.grey[300],
                      child: Icon(Icons.person, color: Colors.white, size: 16),
                    );
                  },
                )
              : Container(
                  color: Colors.grey[300],
                  child: Icon(Icons.person, color: Colors.white, size: 16),
                ),
        ),
      );
    }

    List<Widget> avatarWidgets = [];
    List<String> displayAvatars = group.displayAvatars;
    int displayCount = displayAvatars.length > 4 ? 3 : displayAvatars.length;

    for (int i = 0; i < displayCount; i++) {
      avatarWidgets.add(
        Positioned(
          left: i * 20.0,
          child: Container(
            width: 32,
            height: 32,
            decoration: BoxDecoration(
              shape: BoxShape.circle,
              border: Border.all(color: Colors.white, width: 2),
            ),
            child: ClipOval(
              child: displayAvatars[i].isNotEmpty
                  ? Image.asset(
                      displayAvatars[i],
                      fit: BoxFit.cover,
                      errorBuilder: (context, error, stackTrace) {
                        return Container(
                          color: Colors.grey[300],
                          child: Icon(Icons.person, color: Colors.white, size: 16),
                        );
                      },
                    )
                  : Container(
                      color: Colors.grey[300],
                      child: Icon(Icons.person, color: Colors.white, size: 16),
                    ),
            ),
          ),
        ),
      );
    }

    if (group.totalMemberCount > 4) {
      avatarWidgets.add(
        Positioned(
          left: displayCount * 20.0,
          child: Container(
            width: 32,
            height: 32,
            decoration: BoxDecoration(
              shape: BoxShape.circle,
              color: Colors.grey[400],
              border: Border.all(color: Colors.white, width: 2),
            ),
            child: Center(
              child: Text(
                '+${group.totalMemberCount - 4}', 
                style: TextStyle(
                  color: Colors.white,
                  fontSize: 10,
                  fontWeight: FontWeight.bold,
                ),
              ),
            ),
          ),
        ),
      );
      displayCount++;
    }

    return Container(
      width: (displayCount * 20.0) + 12, 
      height: 32,
      child: Stack(
        children: avatarWidgets,
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: EdgeInsets.only(bottom: 16),
      child: BouncyButton(
        onTap: onTap,
        child: Container(
          padding: EdgeInsets.all(20),
          decoration: BoxDecoration(
            color: Colors.white,
            borderRadius: BorderRadius.circular(20),
            boxShadow: [
              BoxShadow(
                color: Colors.grey.withOpacity(0.12),
                spreadRadius: 0,
                blurRadius: 12,
                offset: Offset(0, 6),
              ),
              BoxShadow(
                color: Colors.grey.withOpacity(0.06),
                spreadRadius: 0,
                blurRadius: 3,
                offset: Offset(0, 2),
              ),
            ],
          ),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(
                group.name,
                style: AppTextStyles.groupTitle,
                maxLines: 2,
                overflow: TextOverflow.ellipsis,
              ),
              
              SizedBox(height: 12),

              Row(
                children: [
                  Icon(
                    Icons.people_outline,
                    size: 20,
                    color: Colors.grey[600],
                  ),
                  SizedBox(width: 4),
                  Text(
                    '${group.memberCount} members',
                    style: AppTextStyles.caption,
                  ),
                ],
              ),
              
              SizedBox(height: 16),
              
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Container(
                    width: 200,
                    padding: EdgeInsets.symmetric(horizontal: 16, vertical: 12),
                    decoration: BoxDecoration(
                      color: _getStatusBackgroundColor(group.status),
                      borderRadius: BorderRadius.circular(20),
                      border: Border.all(
                        color: _getStatusBorderColor(group.status),
                        width: 1,
                      ),
                    ),
                    child: Center(
                      child: Text(
                        _getStatusText(group.status),
                        style: TextStyle(
                          fontSize: 14,
                          fontWeight: FontWeight.w700,
                          color: _getStatusTextColor(group.status),
                        ),
                      ),
                    ),
                  ),
                  
                  _buildMemberAvatars(),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/split/ui/components/GroupCard.kt
package com.example.split.ui.components

import androidx.annotation.DrawableRes
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PeopleOutline
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.split.model.GroupStatus
import com.example.split.model.SplitGroup
import com.example.split.ui.theme.SplitlyColors

@Composable
fun GroupCard(
    group: SplitGroup,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    BouncyButton(
        onClick = onClick,
        modifier = modifier
            .fillMaxWidth()
            .padding(bottom = 16.dp),
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .shadow(
                    elevation = 10.dp,
                    shape = RoundedCornerShape(20.dp),
                    ambientColor = Color(0x1F9E9E9E),
                    spotColor = Color(0x1F9E9E9E),
                )
                .background(Color.White, RoundedCornerShape(20.dp))
                .padding(20.dp),
        ) {
            Text(
                text = group.name,
                style = SplitlyTextStyles.groupTitle,
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )
            Spacer(modifier = Modifier.height(12.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(
                    imageVector = Icons.Filled.PeopleOutline,
                    contentDescription = null,
                    tint = Color(0xFF616161),
                    modifier = Modifier.size(20.dp),
                )
                Spacer(modifier = Modifier.width(4.dp))
                Text(
                    text = "${group.memberCount} members",
                    style = SplitlyTextStyles.caption,
                )
            }
            Spacer(modifier = Modifier.height(16.dp))
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically,
            ) {
                StatusPill(status = group.status)
                AvatarStack(group = group)
            }
        }
    }
}

@Composable
private fun StatusPill(status: GroupStatus) {
    val colors = statusColors(status)
    Box(
        modifier = Modifier
            .width(200.dp)
            .background(colors.background, RoundedCornerShape(20.dp))
            .border(1.dp, colors.border, RoundedCornerShape(20.dp))
            .padding(horizontal = 16.dp, vertical = 12.dp),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = status.label,
            color = colors.foreground,
            fontSize = 14.sp,
            fontWeight = FontWeight.Bold,
        )
    }
}

@Composable
private fun AvatarStack(group: SplitGroup) {
    if (group.memberAvatarResIds.isEmpty()) {
        return
    }

    val displayCount = if (group.totalMemberCount > 4) 3 else group.displayAvatarResIds.size
    val visibleItems = if (group.totalMemberCount > 4) displayCount + 1 else displayCount
    val width = (((visibleItems - 1).coerceAtLeast(0) * 20) + 32).dp

    Box(
        modifier = Modifier
            .width(width)
            .height(32.dp),
    ) {
        group.displayAvatarResIds.take(displayCount).forEachIndexed { index, avatarResId ->
            AvatarImage(
                avatarResId = avatarResId,
                modifier = Modifier.offset(x = (index * 20).dp),
            )
        }

        if (group.totalMemberCount > 4) {
            Box(
                modifier = Modifier
                    .offset(x = (displayCount * 20).dp)
                    .size(32.dp)
                    .background(Color(0xFF9E9E9E), CircleShape)
                    .border(2.dp, Color.White, CircleShape),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    text = "+${group.totalMemberCount - 4}",
                    color = Color.White,
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                )
            }
        }
    }
}

@Composable
private fun AvatarImage(
    @DrawableRes avatarResId: Int,
    modifier: Modifier = Modifier,
) {
    Image(
        painter = painterResource(id = avatarResId),
        contentDescription = null,
        modifier = modifier
            .size(32.dp)
            .clip(CircleShape)
            .border(2.dp, Color.White, CircleShape),
        contentScale = ContentScale.Crop,
    )
}

private val GroupStatus.label: String
    get() = when (this) {
        GroupStatus.Active -> "Active"
        GroupStatus.PaymentProcess -> "Bill Sharing"
        GroupStatus.Done -> "Done"
    }

private data class StatusColors(
    val foreground: Color,
    val background: Color,
    val border: Color,
)

private fun statusColors(status: GroupStatus): StatusColors {
    val foreground = when (status) {
        GroupStatus.Active -> SplitlyColors.Active
        GroupStatus.PaymentProcess -> SplitlyColors.OnProcess
        GroupStatus.Done -> SplitlyColors.Done
    }
    return StatusColors(
        foreground = foreground,
        background = foreground.copy(alpha = 0.1f),
        border = foreground.copy(alpha = 0.3f),
    )
}
